File: modules/apps/object/object-admin-rest-client/src/main/java/com/liferay/object/admin/rest/client/dto/v1_0/ObjectLayoutBox.java
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.object.admin.rest.client.dto.v1_0;

import com.liferay.object.admin.rest.client.function.UnsafeSupplier;
import com.liferay.object.admin.rest.client.serdes.v1_0.ObjectLayoutBoxSerDes;

import jakarta.annotation.Generated;

import java.io.Serializable;

import java.util.Map;
import java.util.Objects;

/**
 * @author Javier Gamarra
 * @generated
 */
@Generated("")
public class ObjectLayoutBox implements Cloneable, Serializable {

	public static ObjectLayoutBox toDTO(String json) {
		return ObjectLayoutBoxSerDes.toDTO(json);
	}

	public Boolean getCollapsable() {
		return collapsable;
	}

	public void setCollapsable(Boolean collapsable) {
		this.collapsable = collapsable;
	}

	public void setCollapsable(
		UnsafeSupplier<Boolean, Exception> collapsableUnsafeSupplier) {

		try {
			collapsable = collapsableUnsafeSupplier.get();
		}
		catch (Exception e) {
			throw new RuntimeException(e);
		}
	}

	protected Boolean collapsable;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public void setId(UnsafeSupplier<Long, Exception> idUnsafeSupplier) {
		try {
			id = idUnsafeSupplier.get();
		}
		catch (Exception e) {
			throw new RuntimeException(e);
		}
	}

	protected Long id;

	public Map<String, String> getName() {
		return name;
	}

	public void setName(Map<String, String> name) {
		this.name = name;
	}

	public void setName(
		UnsafeSupplier<Map<String, String>, Exception> nameUnsafeSupplier) {

		try {
			name = nameUnsafeSupplier.get();
		}
		catch (Exception e) {
			throw new RuntimeException(e);
		}
	}

	protected Map<String, String> name;

	public ObjectLayoutRow[] getObjectLayoutRows() {
		return objectLayoutRows;
	}

	public void setObjectLayoutRows(ObjectLayoutRow[] objectLayoutRows) {
		this.objectLayoutRows = objectLayoutRows;
	}

	public void setObjectLayoutRows(
		UnsafeSupplier<ObjectLayoutRow[], Exception>
			objectLayoutRowsUnsafeSupplier) {

		try {
			objectLayoutRows = objectLayoutRowsUnsafeSupplier.get();
		}
		catch (Exception e) {
			throw new RuntimeException(e);
		}
	}

	protected ObjectLayoutRow[] objectLayoutRows;

	public Integer getPriority() {
		return priority;
	}

	public void setPriority(Integer priority) {
		this.priority = priority;
	}

	public void setPriority(
		UnsafeSupplier<Integer, Exception> priorityUnsafeSupplier) {

		try {
			priority = priorityUnsafeSupplier.get();
		}
		catch (Exception e) {
			throw new RuntimeException(e);
		}
	}

	protected Integer priority;

	public Type getType() {
		return type;
	}

	public String getTypeAsString() {
		if (type == null) {
			return null;
		}

		return type.toString();
	}

	public void setType(Type type) {
		this.type = type;
	}

	public void setType(UnsafeSupplier<Type, Exception> typeUnsafeSupplier) {
		try {
			type = typeUnsafeSupplier.get();
		}
		catch (Exception e) {
			throw new RuntimeException(e);
		}
	}

	protected Type type;

	@Override
	public ObjectLayoutBox clone() throws CloneNotSupportedException {
		return (ObjectLayoutBox)super.clone();
	}

	@Override
	public boolean equals(Object object) {
		if (this == object) {
			return true;
		}

		if (!(object instanceof ObjectLayoutBox)) {
			return false;
		}

		ObjectLayoutBox objectLayoutBox = (ObjectLayoutBox)object;

		return Objects.equals(toString(), objectLayoutBox.toString());
	}

	@Override
	public int hashCode() {
		String string = toString();

		return string.hashCode();
	}

	public String toString() {
		return ObjectLayoutBoxSerDes.toJSON(this);
	}

	public static enum Type {

		CATEGORIZATION("categorization"), REGULAR("regular");

		public static Type create(String value) {
			for (Type type : values()) {
				if (Objects.equals(type.getValue(), value) ||
					Objects.equals(type.name(), value)) {

					return type;
				}
			}

			return null;
		}

		public String getValue() {
			return _value;
		}

		@Override
		public String toString() {
			return _value;
		}

		private Type(String value) {
			_value = value;
		}

		private final String _value;

	}

}

File: modules/apps/object/object-admin-rest-client-js/src/models/ObjectLayoutBox.ts
/**
 * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

			import {ObjectLayoutRow} from './ObjectLayoutRow';

/**
 * @author Javier Gamarra
 * @generated
 */

	export class ObjectLayoutBox {
			"collapsable"?: boolean;
			"id"?: number;
			"name"?: {[key: string]: string;};
			"objectLayoutRows"?: Array<ObjectLayoutRow>;
			"priority"?: number;
			"type"?: 'categorization' | 'regular' | 'seo';

		static "discriminator": string | undefined = undefined;

	static "attributeTypeMap": Array<{
		baseName: string;
		name: string;
		type: string;
	}> = [
		{
			baseName: "collapsable",
			name: "collapsable",
			type: "boolean",
		},
		{
			baseName: "id",
			name: "id",
			type: "number",
		},
		{
			baseName: "name",
			name: "name",
			type: "{[key: string]: string;}",
		},
		{
			baseName: "objectLayoutRows",
			name: "objectLayoutRows",
			type: "Array<ObjectLayoutRow>",
		},
		{
			baseName: "priority",
			name: "priority",
			type: "number",
		},
		{
			baseName: "type",
			name: "type",
			type: "'categorization' | 'regular' | 'seo'",
		},
		];

		static getAttributeTypeMap() {
				return ObjectLayoutBox.attributeTypeMap;
		}
	}
